pkg/issuer/acme/dns/util/wait.go

```diff
@@ -77,19 +78,28 @@ func updateDomainWithCName(r *dns.Msg, fqdn string) string {
 }
 
 // checkDNSPropagation checks if the expected TXT record has been propagated to all authoritative nameservers.
-func checkDNSPropagation(fqdn, value string, nameservers []string) (bool, error) {
-	// Initial attempt to resolve at the recursive NS
-	r, err := dnsQuery(fqdn, dns.TypeTXT, nameservers, true)
-	if err != nil {
-		return false, err
-	}
-	if r.Rcode == dns.RcodeSuccess {
-		fqdn = updateDomainWithCName(r, fqdn)
-	}
+func checkDNSPropagation(fqdn, value string, nameservers []string,
+	useAuthoritative bool) (bool, error) {
+	var authoritativeNss []string
+	if useAuthoritative {
```

Comment:
Can we flip this if/else to be more like:

```
if !useAuthoritative {
    return checkAuthoritativeNss(fqdn, value, nameservers)
}

// Initial attempt to resolve at the recursive NS
r, err := dnsQuery(fqdn, dns.TypeTXT, nameservers, true)
...
```

It'll mean less line changes in the diff, and saves a golint warning! 😄